project/annotations/models.py

```diff
@@ -32,6 +32,16 @@ def label_code(self):
             global_label=self.label, labelset=self.source.labelset)
         return local_label.code
 
+    def save(self, *args, **kwargs):
+        super(Annotation, self).save(*args, **kwargs)
+        # The image's annotation status may need updating.
+        self.image.save()
+
+    def delete(self, *args, **kwargs):
+        super(Annotation, self).delete(*args, **kwargs)
+        # The image's annotation status may need updating.
+        self.image.save()
```

Comment:
Did you test this? Can you access `self` after you deleted it?

Reply:
Good question - that unintuitive-ness hadn't occurred to me. But it does indeed work. After the `super` call, `self.id` is None, but `self.image` is still available. It's tested in `images.tests.test_image_detail_actions.DeleteAnnotationsTest.test_last_annotation_field_updated`.